catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorIT.java

```diff
@@ -0,0 +1,264 @@
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Component;
+import org.codice.ddf.catalog.content.monitor.configurators.KeystoreTruststoreConfigurator;
+import org.codice.ddf.catalog.content.monitor.features.CamelFeatures;
+import org.codice.ddf.catalog.content.monitor.features.CxfFeatures;
+import org.codice.ddf.catalog.content.monitor.features.KarafSpringFeatures;
+import org.codice.ddf.catalog.content.monitor.util.BundleInfo;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.stubbing.Answer;
+import org.ops4j.pax.exam.Option;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+
+import com.google.common.io.Files;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.content.operation.CreateStorageRequest;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.security.service.SecurityManager;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class ContentDirectoryMonitorIT extends AbstractComponentTest {
+
+    @Inject
+    private CamelContext camelContext;
+
+    @Inject
+    private ContentDirectoryMonitor contentDirectoryMonitor;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private String directoryPath;
+
+    private CatalogFramework catalogFramework;
+
+    @Before
+    public void setup() throws IOException {
+        directoryPath = temporaryFolder.getRoot()
+                .getCanonicalPath();
+
+        SecurityManager securityManager = mock(SecurityManager.class);
+        registerService(securityManager, SecurityManager.class);
+
+        catalogFramework = mock(CatalogFramework.class);
+        registerService(catalogFramework, CatalogFramework.class);
+    }
+
+    @Test
+    public void testCamelContext() {
+        Component component = camelContext.getComponent("content");
+        assertThat(component, notNullValue());
+    }
+
+    @Test
+    public void testInPlaceMonitoring()
+            throws IOException, InterruptedException, SourceUnavailableException, IngestException {
+        updateContentDirectoryMonitor(directoryPath, ContentDirectoryMonitor.IN_PLACE);
+
+        File file = createTestFile(directoryPath);
+
+        ArgumentCaptor<CreateStorageRequest> createStorageRequest = ArgumentCaptor.forClass(
+                CreateStorageRequest.class);
+        verify(catalogFramework).create(createStorageRequest.capture());
+        ContentItem result = createStorageRequest.getValue()
+                .getContentItems()
+                .get(0);
+
+        assertThat(result.getFilename(), is(file.getName()));
```

Comment:
❗️ Since our goal here is to verify that the directory monitor always creates the proper content item, we should verify _all_ the `ContentItem` fields, not just the file name.